### PR TITLE
Initial python3 support with FreeCAD backend

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
Just a start. not everything works.

Partially solves #203

### CODE REVIEW

**Comment by jmwright** [comment]
@adam-urbanczyk Thank you for getting this started. Are you going to be modifying this PR over time, or should we review and merge in stages, with the current state being the first stage?

File: cadquery/freecad_impl/__init__.py [hunk 1]
```diff
@@ -141,4 +141,4 @@ def _fc_path():
         import FreeCAD
 
 # logging
-import console_logging
+from . import console_logging
```
Comment (fragmuffin) [inline cadquery/freecad_impl/__init__.py]: good catch, I've really got to pay more attention when I'm importing internal modules.

**Comment by adam-urbanczyk** [comment]
I'll at least add a instruction how to run with python3. NB: there are some tests failing though. CQGI + something kernel related. I think the latter might be FreeCAD related.

**Comment by jmwright** [comment]
@adam-urbanczyk Sounds good. Once the instructions are in place I'll take a look at what's going on with CQGI.

**Comment by adam-urbanczyk** [comment]
@jmwright  I think this is it for now, so you can merge.  To test with python3 follow instructions below.

To create a suitable env run: 

`conda env create -n cq-fc-test -f conda-py3-freecad.yml`

activate it, run:

` export FREECAD_LIB=location_of_freecad_lib`

and let me know what is your result of

`python runtests.py`.

I am at `FAILED (failures=7, errors=5)`

**Comment by jmwright** [comment]
@adam-urbanczyk Is the `conda-py3-freecad.yml` file supposed to be in the repo? I don't see it anywhere. I do see `conda_build_config.yaml` though.

**Comment by adam-urbanczyk** [comment]
@jmwright  are you looking at master (it is not default on my repo)?

I do see it here: https://github.com/dcowden/cadquery/pull/207/commits/0086871aa236eac281b0c0397391551c60b8222d

**Comment by jmwright** [comment]
> are you looking at master (it is not default on my repo)?

@adam-urbanczyk That was it. I was on the default branch. I had assumed that a clone would automatically give me master. Good to know that's now always true.

**Comment by jmwright** [comment]
I've got the environment built, and have done the following export.
```bash
export FREECAD_LIB=/home/jwright/anaconda2/pkgs/freecad-0.17-py36_occt7.1.0_5/lib/FreeCAD.so
```
However, when I try to run the tests, I get the following error.
```bash
$ python ./runtests.py
Traceback (most recent call last):
  File "/home/jwright/Downloads/repos/cadquery/cadquery/freecad_impl/__init__.py", line 137, in <module>
    import FreeCAD
ModuleNotFoundError: No module named 'FreeCAD'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./runtests.py", line 3, in <module>
    from tests import *
  File "/home/jwright/Downloads/repos/cadquery/tests/__init__.py", line 1, in <module>
    from cadquery import *
  File "/home/jwright/Downloads/repos/cadquery/cadquery/__init__.py", line 2, in <module>
    from .freecad_impl.geom import Plane,BoundBox,Vector,Matrix,sortWiresByBuildOrder
  File "/home/jwright/Downloads/repos/cadquery/cadquery/freecad_impl/__init__.py", line 141, in <module>
    import FreeCAD
ModuleNotFoundError: No module named 'FreeCAD'
```
I can think of a few ways to fix that, but I must have missed something somewhere. Am I pointing at the wrong FreeCAD library file?

**Comment by adam-urbanczyk** [comment]
@jmwright  this is what I have:
```
echo $FREECAD_LIB
/home/adam/.virtualenvs/anaconda2/envs/cq-freecad/lib
```

You need to point to the `lib` dir of your anaconda env

**Comment by jmwright** [comment]
Here's the location on my system.
```bash
/home/jwright/anaconda2/envs/cq-fc-test/lib/
```
Here's the result, looks the same as yours.
```
FAILED (failures=7, errors=5)
```
I'm looking into the CQGI errors.

**Pull request merged**